### ISSUE TITLE
Cannot use a scalar value as an array ErrorException  v1.4.0

### ISSUE THREAD

**Issue by Flightfreak** [mailjet/mailjet-apiv3-php#143]
With version  v1.4.0 we experience following `ErrorException`: Cannot use a scalar value as an array
`/var/www/skylegs.com/releases/20181023204356/vendor/mailjet/mailjet-apiv3-php/src/Mailjet/Client.php`

Tested on php7.1-fpm. Any reports on this yet?

**Comment by chandon** [comment]
see #144 and #142 
Don't upgrade to 1.4.0

**Comment by ExnoCell** [comment]
also the basic authentication with key and secret is broken. The following function:

```
private function _isBasicAuthenticationRequired($key, $secret)
    {        
        return is_string($key) && is_string($secret) && empty($secret);
    }
```

returns always false.

I think this must be:

```
private function _isBasicAuthenticationRequired($key, $secret)
    {        
        return is_string($key) && is_string($secret) && !empty($secret);
    }
```

**Comment by xtrembaker** [comment]
+1 (même erreur), j'ai rollback sur 1.3

**Comment by mskochev** [comment]
Should be solved now in v1.4.1 of the wrapper. Please let me know if the problem persists.
